resources/setup-spin

```diff
@@ -0,0 +1,128 @@
+#!/bin/sh
+# This script uses sh instead of bash to be compatible with as many distros as possible.
+
+# The script is located in the Rancher Desktop resources/ directory.
+resources_dir=$(dirname "$0")
+
+# We run setup-spin in the rancher-desktop distro to setup spin on the Win32 host.
+if [ "$WSL_DISTRO_NAME" = "rancher-desktop" ]; then
+  app_data_dir=$(/bin/wslpath "$(powershell.exe -Command "Write-Output \${Env:LOCALAPPDATA}")" | tr -d "\r")
+  system_root=$(/bin/wslpath "$(powershell.exe -Command "Write-Output \${Env:SystemRoot}")" | tr -d "\r")
+  spin="${resources_dir}/win32/bin/spin.exe"
+elif [ "$(uname)" = "Linux" ]; then
+  app_data_dir="${XDG_DATA_HOME:-$HOME/.local/share}"
+  spin="${resources_dir}/linux/bin/spin"
+else
+  app_data_dir="${HOME}/Library/Application Support"
+  spin="${resources_dir}/darwin/bin/spin"
+fi
+
+if [ ! -x "$spin" ]; then
+  echo "Cannot execute '${spin}' (or does not exist)"
+  exit 1
+fi
+
+spin_dir="${app_data_dir}/spin"
+
+# shellcheck disable=SC2012 # Using `ls` is fine
```

Comment:
Might be good to extract that to a `is_nonempty_dir()` for clarity.

Reply:
The whole block can become a `assert_dir_is_empty` function.